packages/core-data/src/actions.js

```diff
@@ -798,6 +798,59 @@ export const saveEditedEntityRecord =
 		return await dispatch.saveEntityRecord( kind, name, record, options );
 	};
 
+/**
+ * Action triggered to update only specified properties for the entity.
+ *
+ * @param {string} kind        Kind of the entity.
+ * @param {string} name        Name of the entity.
+ * @param {Object} recordId    ID of the record.
+ * @param {Array}  itemsToSave List of entity properties or property paths to save.
+ * @param {Object} options     Saving options.
+ */
+export const __experimentalUpdateSpecifiedEntityEdits =
```

Comment:
Just noting that we shouldn't be adding new experimental APIs anymore, we should add private APIs instead.

Reply:
Also it's not really clear what this API offers in addition to what saveEntityRecord does?

Reply:
`saveEntityRecord` saves all the changes to the database. `__experimentalUpdateSpecifiedEntityEdits` saves only the specified edits.


let's use the globalStyles post as example.

- Database has this content:
```json
{ "a": "a"}
```

- Client-side entity has:
```json
{ "a": "a", "b":"b", "c":"c" }
```

- We want to persist to the database only the following JSON without changing the client side entity:
```json
{ "a": "a", "c":"c" }
```

## Examples:


**Using `saveEntityRecord`**

```
saveEntityRecord( 'root', 'post-type', 1, { "a": "a", "b":"b", "c":"c" } )
```

`{ "a": "a", "b":"b", "c":"c" }` is persisted to the database.



**Using `__experimentalUpdateSpecifiedEntityEdits`:**
```js
__experimentalUpdateSpecifiedEntityEdits( 'root', 'post-type', 1, [  'c'  ] );
```
 `{ "a": "a", "c":"c" }` is to the database.



Using
```js
__experimentalSaveSpecifiedEntityEdits( 'root', 'post-type', 1, [  'c'  ] );
```
`{"c":"c" }` is to the database without changing the client-side entity.


---
I added these examples to the PR's description.

Probably the behavior added in this PR for `__experimentalUpdateSpecifiedEntityEdits` was the one originally thought for `__experimentalSaveSpecifiedEntityEdits`. In that case, we can move the additions to that function instead of adding a new one.

Reply:
> __experimentalUpdateSpecifiedEntityEdits( 'root', 'post-type', 1, [  'c'  ] );

But in this case, can't we just send `saveEntityRecord( 'root', 'post-type', 1, { "c":"c" } )` directly? In other words, can't we retrieve the edits that we want to save and just pass them (or not even mark them as edits). I'm pretty sure we do this already in other places already.

Reply:
`__experimentalSaveSpecifiedEntityEdits` to be honest I don't know much about this function, I need to check the history.

Reply:
> Probably the behavior added in this PR for __experimentalUpdateSpecifiedEntityEdits was the one originally thought for __experimentalSaveSpecifiedEntityEdits. In that case, we can move the additions to that function instead of adding a new one.

This is actually very possible and in which case, we might have a bug in the places where `__experimentalSaveSpecifiedEntityEdits` is used (saving specific site properties)

Reply:
> But in this case, can't we just send saveEntityRecord( 'root', 'post-type', 1, { "c":"c" } ) directly? 

This would create a new post revision only with `{ "c":"c" }` content in the database.

That's how it's working right now and what's producing the bug reported in https://github.com/WordPress/gutenberg/issues/60343

Reply:
> This would create a new post revision only with { "c":"c" } content in the database instead of the desired { "a": "a", "c":"c" }.

I guess that it's possible that it's an endpoint issue then because if I'm not wrong all of the endpoints in Core work as "PATCH" (maybe not the global styles one, we need to check)

Reply:
It is also why we didn't see the issue before with `__experimentalSaveSpecifiedEntityEdits` (we never used it with the global styles endpoint before)

Reply:
Oh so you're actually trying to send only `settings.typography.fontFamilies` and not `settings`. That is going to override the whole "settings" object of course. So IMO, you should just send the update `settings` object entirely. 

The endpoints don't do "nested patches", only top level.

Reply:
> This would create a new post revision only with { "c":"c" } content in the database instead of the desired { "a": "a", "c":"c" }.
> 
> I don't think this is true for top level keys.

That's correct in my testing for the global-styles endpoint, and true generally across the REST API. `styles` are preserved here when a font is activated because it's a different top level key, but `settings` are reset (without this PR) because they're empty except for `settings.typography.fontFamilies`.